From: Xuewen Wang <wangxuewen@kylinos.cn>
Subject: tools/mm/slabinfo: remove dead assignment in get_obj_and_str()
Date: Mon, 18 May 2026 14:21:58 +0800

The assignment `x = NULL` sets the local parameter variable instead of
`*x`, which is a no-op since `*x` was already set to NULL on the line
above.  Remove the dead assignment.

Link: https://lore.kernel.org/20260518062159.80664-3-wangxuewen@kylinos.cn
Signed-off-by: Xuewen Wang <wangxuewen@kylinos.cn>
Reviewed-by: Andrew Morton <akpm@linux-foundation.org>
Reviewed-by: SeongJae Park <sj@kernel.org>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/mm/slabinfo.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

--- a/tools/mm/slabinfo.c~tools-mm-slabinfo-remove-dead-assignment-in-get_obj_and_str
+++ a/tools/mm/slabinfo.c
@@ -193,10 +193,9 @@ static unsigned long get_obj_and_str(con
 
 	*x = NULL;
 
-	if (!read_obj(name)) {
-		x = NULL;
+	if (!read_obj(name))
 		return 0;
-	}
+
 	result = strtoul(buffer, &p, 10);
 	while (*p == ' ')
 		p++;
_
